### ISSUE TITLE
Floating menu: Keyboard Navigation to Enter and Exit

### ISSUE THREAD

**Issue by miina** [swissspidy/web-stories-wp#10872]
## Feature Description
Generally, the floating menu is using roving index keyboard navigation as other context menus (navigating with arrow keys, tab moves to the next section), however, since we have a text input in the menu, this doesn't work exactly as expected.

Some thoughts from @barklund:

> Google Docs and others do this by allowing you to use arrow keys to move between the menu elements, including moving focus to an input field. But the input field acts like a menu item button, so you have to press enter when focus is on the input field menu item to actually go into typing in the input field. When there, arrow keys work as normal in a numeric input field, moving the cursor and stepping the value. Then you can press escape to move back out of the input field, moving focus back to the menu item and you can again use arrow keys to move between menu items.
>
> Note in this gif how the Google Docs menu bar works. I can use arrow keys to move between the elements in the menu, not tab. Using arrow keys, i just move over the font size input as if it was a button. Only once I press enter on the "input button", do I get to interact with the input.
>
> ![gdocs-menu-bar-input](https://user-images.githubusercontent.com/637548/157508949-0148ba2c-94bb-4ff8-a4b5-0b0e8d7b0710.gif)
>
> Google Docs then decides to always move keyboard focus back to the document whenever you select something in the menu (even when doing so with the keyboard), so as soon as I press enter inside the input field to commit my choice of font size, focus moves to the document.
>
> I'm not sure if we want to do the same. We could alternatively move focus back to the "input button" rather than the input itself, so arrow keys would once again work. We could then have <key>esc</key> work in the same way except that any input would be reset to the original state before moving focus back to the input button.
>
> Another related problem is, that the focus management handled by the menu does not follow actual focus. If I press a button and then press arrow keys to move to the next button, the arrow keys will take me to a different button because the arrow key focus and actual focus are not in sync.

See also the [comment thread and thoughts](https://github.com/GoogleForCreators/web-stories-wp/pull/10859#issuecomment-1063266518) from @BrookeGraham 

The current idea (as of 10th of March) is to change the role of the menu to "toolbar" and implement the relevant parts from [its spec](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/toolbar_role).

**Comment by maxyinger** [comment]
@BrittanyIRL  Just to clarify, the expected scope of this ticket is:
- can `enter` while element is both selected & focused to access the toolbar (for all element types)
- can `esc` to return focus back to the selected element while in the toolbar
- toolbar has revolving keyboard nav, where pressing `right` on the last item returns you to the first, (only pressing `enter` on the `more` button and `esc` will navigate you out of the floating menu)
- must `enter` into and `esc` out of all text inputs to interact within the toolbar (fixes arrow navigation)
- floating menu should be `role=toolbar`
- do we want to remove this from the tab flow if it's not accessed by pressing `enter` on the selected element? ie when tab navigating through the navLayer the floating menu shouldn't receive focus?

Just want to make sure this scope hasn't been documented in other tickets or I'm not missing anything.

**Comment by BrittanyIRL** [comment]
Thank you, @littlemilkstudio . Yes, i think this is the proper acceptance criteria here. Please speak up if you think any of it should be isolated as it's own ticket. I am aware that I bumped this from a 3 to a 5 after we discussed it more in depth last week in relation to the canvas as a whole.

**Comment by maxyinger** [comment]
@BrittanyIRL perfect. thanks! yup, should definitely be a 5. thank you for bumping!

**Comment by BrittanyIRL** [comment]
Since getting the first 2 bullet points listed here done was actually incredibly involved I have separated the additional work to create the revolving keyboard nav (and updating the role type) in its own issue (#11162 ). So this issue will only be solving the enter and exit of floating menu by keyboard.

**Comment by BrittanyIRL** [comment]
@felipebochehin87 this is just part of the focus changes getting updated to impact the whole canvas. So we just need to make sure the following is true: 

- with an element selected & focused, press `ctrl+alt+p` and see that focus goes inside the floating menu
- now press esc and see that focus leaves the floating menu
